sonar-kotlin-checks/src/main/java/org/sonarsource/kotlin/checks/CollectionShouldBeImmutableCheck.kt

```diff
@@ -52,8 +59,25 @@ class CollectionShouldBeImmutableCheck : AbstractCheck() {
             "kotlin.collections.MutableCollection"
         )
 
+    override fun visitClass(klass: KtClass, context: KotlinFileContext) {
+        if (klass.isInterface()) return
+
+        klass
+            .collectDescendantsOfType<KtNamedFunction>(::notAnInterface) { true }
+            .forEach { reportNamedFunction(it, context) }
+    }
```

Comment:
Like this, we would miss any cases that are inside of file-level functions. Like this you can get those too:

```suggestion
    override fun visitKtFile(file: KtFile, context: KotlinFileContext) {
        file
            .collectDescendantsOfType<KtNamedFunction>(::notAnInterface) { true }
            .forEach { reportNamedFunction(it, context) }
    }
```

Reply:
You can put this on file level into the sample file:
```
private fun nonCompliantParameterOnFileLevel(list: MutableList<Int>): Int { // Noncompliant
    return list.reduce { acc, it -> acc + it}
}
```